src/substrait/textplan/converter/PlanPrinterVisitor.cpp

```diff
@@ -7,58 +7,567 @@
+      text << literal.fp64() << "_fp64";
+      break;
+    case ::substrait::proto::Expression::Literal ::kDate: {
+      // TODO -- Format this as a date instead of a delta since an epoch.
+      if (literal.date() >= 0) {
+        text << "\"epoch+" << literal.date() << " days\"_date";
+      } else {
+        text << "\"epoch" << literal.date() << " days\"_date";
+      }
+      break;
+    }
+    case ::substrait::proto::Expression::Literal::kString:
+      text << "\"" << stringEscape(literal.string()) << "\"_string";
+      break;
+    case ::substrait::proto::Expression_Literal::kBinary:
+    case ::substrait::proto::Expression_Literal::kTimestamp:
+    case ::substrait::proto::Expression_Literal::kTime:
+      error_listener_->addError(
+          "Literal of this type not yet supported: " +
+          literal.ShortDebugString());
+      return std::string("UNSUPPORTED_LITERAL_TYPE");
+    case ::substrait::proto::Expression_Literal::kIntervalYearToMonth: {
+      text << "{";
+      bool hasPreviousText = false;
+      if (literal.interval_year_to_month().years() != 0) {
+        text << literal.interval_year_to_month().years() << "years";
+        hasPreviousText = true;
+      }
+      if (literal.interval_year_to_month().months() != 0) {
+        if (hasPreviousText)
+          text << ", ";
+        text << literal.interval_year_to_month().months() << "months";
+      }
+      text << "}_interval_year"; // TODO - Change spec to better name.
+      break;
+    }
+    case ::substrait::proto::Expression_Literal::kIntervalDayToSecond: {
+      text << "{";
+      bool hasPreviousText = false;
+      if (literal.interval_day_to_second().days() != 0) {
+        text << literal.interval_day_to_second().days() << "days";
+        hasPreviousText = true;
+      }
+      if (literal.interval_day_to_second().seconds() != 0) {
+        if (hasPreviousText)
+          text << ", ";
+        text << literal.interval_day_to_second().days() << "seconds";
+        hasPreviousText = true;
+      }
+      if (literal.interval_day_to_second().microseconds() != 0) {
+        if (hasPreviousText)
+          text << ", ";
+        text << literal.interval_day_to_second().days() << "microseconds";
+      }
+      text << "}_interval_day"; // TODO - Change spec to better name.
+      break;
+    }
+    case ::substrait::proto::Expression_Literal::kFixedChar:
+      // TODO - Should the fixedchar length match its use instead?
+      text << "\"" << stringEscape(literal.fixed_char()) << "\"_fixedchar<"
+           << literal.fixed_char().length() << ">";
+      break;
+    case ::substrait::proto::Expression_Literal::kVarChar:
+      text << "\"" << stringEscape(literal.var_char().value()) << "\"_varchar<"
+           << literal.var_char().length() << ">";
+      break;
+    case ::substrait::proto::Expression_Literal::kDecimal: {
+      auto dec = common::Decimal::fromProto(literal.decimal());
+      return dec.toString();
```

Comment:
This doesn't match the format of decimals defined in https://github.com/substrait-io/substrait-cpp/pull/33/files#diff-94fce6a3a4a591e4cd44ad4364294611a3ad6cd117450d90d3888933b5230f7aR96